### PR TITLE
feat(multipaths): [LIT03-1136] optimize calling the same endpoint for…

### PR DESCRIPTION
**Pull request by sirjasiu**
… several json-paths in API merger

BREAKING CHANGE: NEW CONFIG

### CODE REVIEW

File: kong/plugins/api-response-merger/body_transformer.lua [hunk 1]
```diff
@@ -372,26 +383,26 @@ function _M.transform_json_body(keys_to_extend, upstream_body, http_config)
...
-    local resource = resources[resource_key]
+    local resource_index, resource_body = unpack(result)
+    local resource = resources[resource_index]
     local config = resource.config
     -- if response is an array we should use set_in_table_arr
     -- multiple resources
 
     if utils.is_array(upstream_body, 'fast') then
-      local _, err = set_paths(resource_key, upstream_body, resource_body, resource)
+      local _, err = set_paths(resource.config.data_paths, upstream_body, resource_body, resource)
```
Comment (aldor007) [inline kong/plugins/api-response-merger/body_transformer.lua:393]:
```suggestion
      local _, err = set_paths(config.data_paths, upstream_body, resource_body, resource)
```

File: kong/plugins/api-response-merger/body_transformer.lua [hunk 2]
```diff
@@ -372,26 +383,26 @@ function _M.transform_json_body(keys_to_extend, upstream_body, http_config)
...
         return false, create_error_response(err, config.api.url, 200, resource_body)
       end
     else
       if resource_body == nil and config.allow_missing == false then
-        kong.log.err('Missing data for resource ', resource_key, ' api: ', config.api.url, ' res_body: nil')
+        kong.log.err('Missing data for resource ', resource_index, ' api: ', config.api.url, ' res_body: nil')
         return false, err
       end
-      local _, err = set_paths(resource_key, upstream_body, resource_body, resource)
+      local _, err = set_paths(resource.config.data_paths, upstream_body, resource_body, resource)
```
Comment (aldor007) [inline kong/plugins/api-response-merger/body_transformer.lua:403]:
```suggestion
      local _, err = set_paths(config.data_paths, upstream_body, resource_body, resource)
```

File: kong/plugins/api-response-merger/body_transformer.lua [hunk 3]
```diff
@@ -125,14 +125,15 @@ local function get_external_id_and_resource_from_value(dest_resource, value, res
...
+local function set_for_path(data_path, upstream_body, value, resource)
   if value == nil then
     return nil, nil
   end
 
+  local path = data_path.path
   local dest_resource_paths, err = jp.paths(upstream_body, path)
   if err ~= nil then
-    kong.log.error("unable to find paths ", path, " error ", err)
+    kong.log.error("unable to find paths ", data_path.path, " error ", err)
```
Comment (aldor007) [inline kong/plugins/api-response-merger/body_transformer.lua:136]:
```suggestion
    kong.log.error("unable to find paths ", path, " error ", err)
```

**Review by aldor007 — APPROVED** [review]
LGTM
Really good work!

**Review by aldor007 — APPROVED** [review]

**Review by marcin-shoemaker — APPROVED** [review]

**Pull request merged**